docs/internal/release-versions.md

```diff
@@ -0,0 +1,129 @@
+* as a dependency on another piece (e.g. fix in a Chain Core needs corresponding Mac and Windows executables)
+
+We need a consistent, yet simple, versioning scheme that allows both individual updates as well as grouped updates: from major feature releases to smaller, maintenance releases.
+
+We probably do not want to formally commit to specific semantics of the version numbers, such as [SemVer](http://semver.org), but we need to keep all numbers in order, so it’s easier navigate them.
+
+## Proposal
+
+Use the following version format:
+
+    <major>.<minor>.<build>
+
+**Major** and **minor** versions are reserved for the entire product updates usually affecting several pieces of software. Either version can be bumped based on the significance of the update. Individual pieces of software must not use either **major** or **minor** version on their own without synchronization with the entire product line, but instead should use **build** version.
+
+**Build** version must be formatted as dates in the following format: `YYYYMMDD` or omitted entirely. **Major** and **minor** versions are plain integers.
+
+Examples:
+
+* 1.0
+* 1.0.20161024
+* 1.7.20170119
+* 2.3.20170601
+
+Compatibility notes must be specified separately. See [Blockchain Extensibility](../protocol/papers/blockchain-extensibility.md) document for details of blockchain compatibility. SDKs should deprecate old APIs and introduce new ones smoothly and may change entirely with major product releases.
+
+## Build version
+
+The reason to use dates for **build** versions is to avoid frequent collisions across different software pieces that may convey non-existant correlation. E.g. if at some point Java SDK and Mac app both have version `1.3.2`, some may be confused by the identical version. 
+
+On the other hand, dependent releases usually happen on the same day and therefore most likely will have the identical versions, indicating the correlation truthfully. E.g. if Chain Core adds a minor fix in the API, all SDKs may be updated with the same new build number: `1.0.161205`.
+
+## Compatibility
+
+We should strive to keep expectation that minor version updates do not break backward compatibility. E.g. if we need to introduce a completely new SDK, we should bump the major version, or rename the SDK. If we need to retire/improve all APIs, we should add a compatibility layer and diagnostic messages to the SDK without simply removing them.
+
+Do not expect this guide to cover all imaginable scenarios, use common sense and the context to decide the versioning in unusual situations. Also, even in such cases versioning guidelines can be seen not as an unfortunate barrier, but as a useful constraint that may guide us to a better positioning of the product updates and help avoid 
+
+
+## Examples
+
+Suppose, we have the following versions currently shipping:
+
+* Chain Core 1.0
+* Java SDK 1.0
+* Chain Core for Mac 1.0
+* Chain Core Windows Installer 1.0
+
+### Case 1: a bugfix in the SDK
```

Comment:
I'm not sure it makes sense to distinguish between a bugfix and a new feature in this place. What if we add a new feature to the sdk?